packages/flutter/lib/src/widgets/raster_widget.dart

```diff
@@ -0,0 +1,165 @@
+      ..delegate = delegate
+      ..devicePixelRatio = MediaQuery.maybeOf(context)?.devicePixelRatio ?? 1.0;
+  }
+}
+
+/// A delegate used to draw the `ui.Image` representing the rasterized child widget.
+abstract class RasterWidgetDelegate extends ChangeNotifier {
+  /// Called whenever the [image] that represents a [RasterWidget]s child should be painted.
+  ///
+  /// The image is rasterized at the physical pixel resolution and should be scaled down by
+  /// [pixelRatio] to account for device independent pixels.
+  ///
+  /// There is no offset given in this paint method, as the parent is an [OffsetLayer] all
+  /// offsets are `Offset.zero`.
+  ///
+  /// Example:
+  ///
+  /// ```dart
+  /// void paint(PaintingContext context, Rect area, ui.Image image, double pixelRatio) {
+  ///   var src = Rect.fromLTWH(0, 0, image.width, image.height);
+  ///   var dst = Rect.fromLTWH(0, 0, area.width, area.height);
+  ///   context.canvas.drawImageRect(image, src, dst);
+  /// }
+  ///
+  /// ```
+  void paint(PaintingContext context, Rect area, ui.Image image, double pixelRatio);
+
+  /// Called whenever a new instance of the custom painter delegate class is
+  /// provided to the [RenderRasterWidget] object, or any time that a new
+  /// [RasterWidgetDelegate] object is created with a new instance of the custom painter
+  /// delegate class (which amounts to the same thing, because the latter is
+  /// implemented in terms of the former).
+  ///
+  /// If the new instance represents different information than the old
+  /// instance, then the method should return true, otherwise it should return
+  /// false.
+  ///
+  /// If the method returns false, then the [paint] call might be optimized
+  /// away.
+  ///
+  /// It's possible that the [paint] method will get called even if
+  /// [shouldRepaint] returns false (e.g. if an ancestor or descendant needed to
+  /// be repainted). It's also possible that the [paint] method will get called
+  /// without [shouldRepaint] being called at all (e.g. if the box changes
+  /// size).
+  ///
+  /// The `oldDelegate` argument will never be null.
+  bool shouldRepaint(covariant RasterWidgetDelegate oldDelegate);
+}
+
+/// A render object that draws its child as a [ui.Image].
+class RenderRasterWidget extends RenderProxyBox {
+  /// Create a new [RenderRasterWidget].
+  RenderRasterWidget(this._delegate, this._devicePixelRatio);
+
+  /// The device pixel ratio used to create the child raster.
+  double get devicePixelRatio => _devicePixelRatio;
+  double _devicePixelRatio;
+  set devicePixelRatio(double value) {
+    if (value == devicePixelRatio) {
+      return;
+    }
+    _devicePixelRatio = value;
+    markNeedsPaint();
+  }
+
+  /// The delegate used to draw the `ui.Image` representing the rasterized child widget.
+  RasterWidgetDelegate get delegate => _delegate;
+  RasterWidgetDelegate _delegate;
+  set delegate(RasterWidgetDelegate value) {
+    if (!value.shouldRepaint(delegate)) {
+      return;
```

Comment:
Shouldn't this still update the `_delegate`?

Reply:
Done